ambry-commons/src/main/java/com/github/ambry/clustermap/HelixVcrUtil.java

```diff
@@ -314,25 +309,27 @@ public static boolean isSrcDestSync(String srcZkString, String srcClusterName, S
     HelixAdmin destAdmin = new ZKHelixAdmin(destZkString);
     Set<String> destResources = new HashSet<>(destAdmin.getResourcesInCluster(destClusterName));
 
+    boolean status = true;
     for (String resource : srcResources) {
       if (!isPartitionResourceName(resource)) {
-        System.out.println("Resource " + resource + " from src cluster is ignored");
+        logger.error("[HelixSync] Ignore resource {} from source Ambry cluster", resource);
         continue;
       }
-      if (destResources.contains(resource)) {
-        // check if every partition exist.
-        Set<String> srcPartitions = srcAdmin.getResourceIdealState(srcClusterName, resource).getPartitionSet();
-        Set<String> destPartitions = destAdmin.getResourceIdealState(destClusterName, resource).getPartitionSet();
-        for (String partition : srcPartitions) {
-          if (!destPartitions.contains(partition)) {
-            return false;
-          }
+      if (!destResources.contains(resource)) {
+        logger.error("[HelixSync] Resource {} from source Ambry cluster is absent in VCR cluster", resource);
+        status = false;
+      }
+      // check if every partition exist.
+      Set<String> srcPartitions = srcAdmin.getResourceIdealState(srcClusterName, resource).getPartitionSet();
+      Set<String> destPartitions = destAdmin.getResourceIdealState(destClusterName, resource).getPartitionSet();
+      for (String partition : srcPartitions) {
+        if (!destPartitions.contains(partition)) {
+          logger.error("[HelixSync] Partition {} from source Ambry cluster is absent in VCR cluster", partition);
+          status = false;
```

Comment:
We don't need to go through all the inconsistent resources and partition to return false, we can just return false when first inconsistency is detected. What is the purpose of having iterating through all of the inconsistency?

Reply:
the idea is to print out all inconsistencies so the on-call or dev can take appropriate action for all inconsistencies at once.

Reply:
the job is set to run every 1 hr

Reply:
But we have an schedule task to sync them up, so it doesn't really need human intervention. If 1 hour is too long, we can set it to 10 minutes.

Reply:
changed to return on first inconsistency. but this whole code around helix sync-update task looks bad. Why do we have two separate tasks to check for sync and update ? They could be one job. Secondly, why aren't these jobs scheduled in the constructor ? these are being scheduled in helix context which is just horrible.

Reply:
the issue is that we only want the host that have partition 0 to do the sync task so we don't run into any concurrency issues.

Reply:
and separation of check and update task is indeed really weird.